### PR TITLE
[19] As a user, I can search across my reports

### LINKED ISSUE #19: As a user, I can search across my reports
## Why

A user should be able to search across reports using a search bar.

## Acceptance Criteria

1. Have the dashboard pages display a search bar
2. Typing a query and searching should take them to a page with the search results

### PR DESCRIPTION
**Pull request by bterone**
Closes #19 

## What happened

✅: Move header to separate partial.
✅: Add body class name helper for scoping screen CSS.
✅: Add UI for frontend portion of #54 
 
## Insight

Took the `body_class_name/1` from our implementation at zipmex.
 
## Proof Of Work

Screen Shot 2021-05-23 at 10 53 37 AM

### CODE REVIEW

File: lib/g_searcher_web/controllers/search_result_controller.ex [hunk 1]
```diff
@@ -9,7 +9,7 @@ defmodule GSearcherWeb.SearchResultController do
 
     search_results = SearchResults.list_search_results_by_user_id(user_id, params["query"])
 
-    render(conn, "index.html", search_results: search_results)
+    render(conn, "index.html", search_results: Enum.with_index(search_results, 1))
```
Comment (hanam1ni) [inline lib/g_searcher_web/controllers/search_result_controller.ex]: I think we can move `Enum.with_index/2`  to the template, it feels like a view logic.

File: lib/g_searcher_web/templates/report/show.html.eex [hunk 2]
```diff
@@ -52,5 +52,5 @@
 </section>
 
 <section class="report-search-results">
-  <%= render "_search-result-table.html", search_results: @search_results, conn: @conn %>
+  <%= render GSearcherWeb.SharedView, "_search-result-table.html", assigns %>
```
Comment (hanam1ni) [inline lib/g_searcher_web/templates/report/show.html.eex]: How about passing only the needed assigns, `search_results`?

**Review by hanam1ni — APPROVED** [review]

File: lib/g_searcher_web/templates/report/show.html.eex (hunk 2 above)
Comment (hanam1ni) [inline lib/g_searcher_web/templates/report/show.html.eex]: Does this work? so we pass only needed assign.
```suggestion
  <%= render GSearcherWeb.SharedView, "_search-result-table.html", search_results: @search_results %>
```

Comment (bterone) [reply lib/g_searcher_web/templates/report/show.html.eex]: Yes it does, this assigns is different from the `conn.assigns` attribute. This basically is passing everything that we've got from the controller, including the `@conn`. I think I forgot to change this back 😂 

We're using this way on Zipmex. @hanam1ni 😄

Comment (hanam1ni) [reply lib/g_searcher_web/templates/report/show.html.eex]: @bterone I am not sure about the benefit to pass everything to the partial template. Maybe you can give me some insight into this case.

Anyway PR LGTM, approved 👍

**Review by hanam1ni — APPROVED** [review]

File: lib/g_searcher_web/templates/report/show.html.eex (hunk 2 above)
Comment (hanam1ni) [reply lib/g_searcher_web/templates/report/show.html.eex]: If you have time, you can share in #elixir-guild for other developers too 🙌

**Review by andyduong1920 — APPROVED** [review]
LGTM @bterone 💪

**Pull request merged**